ports/ffmpeg/CONTROL

```
@@ -1,6 +1,5 @@
 Source: ffmpeg
-Version: 4.2-8
-Build-Depends: zlib
```

Should we set zlib as a default feature?

---

Previously it was listed as a default feature but ffmpegs configure was silently failing and never actually using it. So not having it included as default is actually consistent with the builds that vcpkg was previously generating

---

By default (non vcpkg builds) ffmpeg auto enables bzlib2, iconv, lzma, sdl2, zlib libraries. So if we were to auto enable libraries such as zlib then those others would also be on the list if we are trying to be consistent with that.

---

OK to fully cover the whole zlib depends issue and hopefully help others understand and avoid confusion in the long term here is a little FYI:

vcpkg's cmake file just sets some build options and then calls ffmpegs built in configure/make based build script. By default, unless told to do otherwise, ffmpegs build script will auto test for the availability and **usability** of some (not all) of its dependencies (zlib, bzip2, iconv, lzma, sdl2 and some other platform dependent options) and enable them. 

When building, if zlib was installed prior to ffmpeg then ffmpegs build script will detect its available and try and use it. But on windows builds due to the unistd.h error that this PR fixes it would be detected as **not usable** and would therefore not be enabled.

My guess is that the reason that zlib was listed as a build dependency was that someone built ffmpeg on non-windows using static linking (dynamic linking works fine). In this case there is a valid unistd.h and as such zlib was detected as **usable** by ffmpegs build script and automatically included. This would cause unforeseen link errors about missing zlib and give the incorrect impression that zlib was a required dependency. However this could also happen for any of the other dependencies that ffmpeg may auto enable (listed above) but only the zlib case was previously handled meaning all other cases would fail as well.

With this patch, vcpkgs cmake script is updated so that it now takes direct control over which external dependencies are used and prevents ffmpegs build script from attempting to auto enable any of them by passing it the appropriate configure options. As such there are now no hidden auto enabled dependencies and the only dependencies used are the ones vcpkg users explicitly enable.